dapp/src/lib/interfaces/routes.interfaces.ts

```diff
@@ -1,11 +1,11 @@
 // Copyright (c) 2025 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-export interface ProtectedRoute {
-    title: string;
+export interface Route {
+    title?: string;
     path: string;
-    icon?: (props: React.SVGProps<SVGSVGElement>) => React.JSX.Element;
-    id: string;
+    icon?: JSX.Element;
+    id?: string;
 }
 
-export interface PublicRoute extends Pick<ProtectedRoute, 'path'> {}
+export interface PublicRoute extends Pick<Route, 'path'> {}
```

Comment:
What do you think about doing it like this instead?
```suggestion
export type Route = {
	type: 'protected';
    title: string;
path: string;
    icon?: JSX.Element;
    id: string;
} | {
	type: 'public';
	path: string
};
```

Reply:
or like this so we only rely on a boolean? @marc2332 
```tsx
type PublicRoute = {
    path: string;
    isProtected?: false;
};

type ProtectedRoute = {
    path: string;
    isProtected: true;
    icon?: JSX.Element;
    id: string;
};

export type Route = PublicRoute | ProtectedRoute;

const ROUTES: Route[] = [
    {
        path: '/',
    },
    {
        path: '/dashboard',
        isProtected: true,
        id: 'dashboard',
    },
];
```

Reply:
Ah yep, that works too!